libraries/abstractions/platform/freertos/iot_network_freertos.c

```diff
@@ -595,23 +595,27 @@ size_t IotNetworkAfr_ReceiveUpto( void * pConnection,
         pNetworkConnection->bufferedByteValid = false;
     }
 
-    /* Block and wait for incoming data. */
-    socketStatus = SOCKETS_Recv( pNetworkConnection->socket,
-                                 pBuffer + bytesReceived,
-                                 bufferSize - bytesReceived,
-                                 0 );
-
-    if( socketStatus <= 0 )
-    {
-        IotLogError( "Error %ld while receiving data.", ( long int ) socketStatus );
-    }
-    else
+    if( bufferSize - bytesReceived > 0 )
     {
-        bytesReceived += ( size_t ) socketStatus;
-        IotLogDebug( "Successfully received %lu bytes.",
-                     ( unsigned long ) bytesReceived );
+        /* Block and wait for incoming data. */
```

Comment:
This comment assumes the socket is a blocking one?

Reply:
That's correct, the sockets are blocking.